src/layout/monitor.rs

```diff
@@ -575,6 +575,18 @@ impl<W: LayoutElement> Monitor<W> {
         self.clean_up_workspaces();
     }
 
+    // returns a rectangle describing where the focused tile is located in visual coordinates
+    pub fn active_tile_visual_rectangle(&mut self) -> Option<Rectangle<i32, Logical>> {
+        let workspace = self.active_workspace();
+        if workspace.columns.is_empty() {
+            return None;
+        }
+        let col = &mut workspace.columns[workspace.active_column_idx];
+        let tile_size = col.tiles[col.active_tile_idx].tile_size();
```

Comment:
This sounds like it should be inside the function on Workspace.

Reply:
fixed in commit [123c2f0](https://github.com/YaLTeR/niri/pull/245/commits/123c2f040e1891c7ce5d3d7c29d497f48da788cd)